render-ws-java-client/src/main/java/org/janelia/render/client/newsolver/StackAlignmentErrorClient.java

```diff
@@ -2,85 +2,77 @@
 import mpicbg.models.Model;
 import mpicbg.models.NoninvertibleModelException;
 import mpicbg.models.PointMatch;
 import mpicbg.models.RigidModel2D;
-import mpicbg.models.TranslationModel2D;
-
 import org.janelia.alignment.match.CanvasMatches;
 import org.janelia.alignment.match.MatchCollectionId;
 import org.janelia.alignment.match.OrderedCanvasIdPair;
+import org.janelia.alignment.match.OrderedCanvasIdPairWithValue;
 import org.janelia.alignment.spec.Bounds;
+import org.janelia.alignment.spec.ResolvedTileSpecCollection;
 import org.janelia.alignment.spec.ResolvedTileSpecsWithMatchPairs;
 import org.janelia.alignment.spec.TileSpec;
 import org.janelia.alignment.spec.stack.StackId;
 import org.janelia.alignment.spec.stack.StackMetaData;
+import org.janelia.alignment.util.NeuroglancerUtil;
 import org.janelia.alignment.util.ResidualCalculator;
 import org.janelia.render.client.ClientRunner;
 import org.janelia.render.client.RenderDataClient;
+import org.janelia.render.client.newsolver.errors.AlignmentErrors;
+import org.janelia.render.client.newsolver.errors.AlignmentErrors.MergingMethod;
 import org.janelia.render.client.newsolver.solvers.WorkerTools;
 import org.janelia.render.client.parameter.CommandLineParameters;
 import org.janelia.render.client.parameter.MatchCollectionParameters;
 import org.janelia.render.client.parameter.RenderWebServiceParameters;
-import org.janelia.render.client.solver.StabilizingAffineModel2D;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 public class StackAlignmentErrorClient {
 
+	public enum ErrorMetric {
+		GLOBAL_LOCAL_DIFFERENCE,
+		RMSE
+	}
+
 	public static class Parameters extends CommandLineParameters {
 		@ParametersDelegate
 		private final RenderWebServiceParameters renderParams = new RenderWebServiceParameters();
 		@ParametersDelegate
 		private final MatchCollectionParameters matchParams = new MatchCollectionParameters();
 		@Parameter(names = "--stack", description = "Stack for which to compute errors", required = true)
 		private String stack;
-		@Parameter(names = "--fileName", description = "Name of file to write pairwise errors to (default: errors_<stack>.json.gz)")
-		private String fileName = null;
 		@Parameter(
-				names = "--calculateResiduals",
-				description = "Calculate root mean squared error for each pair of tiles (default: calculate Preibisch error)",
-				arity=0)
-		private boolean calculateResiduals;
-
-		// TODO: define enum for different error types (Preibisch, RMSE, ...) and use that instead of boolean flag
-
-		public String getFileName() {
-			if (fileName == null) {
-				fileName = "errors_" + stack + ".json.gz";
-			}
-			return fileName;
-		}
+				names = "--errorMetric",
+				description = "Error metric to use for computing errors (default: GLOBAL_LOCAL_DIFFERENCE)")
```

Comment:
There is no need to put default values in the description because utility automatically does that.

Reply:
Oh, thanks for clarifying! I didn't know (and apparently never noticed). I'll remove default value description in the future if they can be inferred from context.